### PR TITLE
Fixing np_random att visibility in core.Task class

### PR DESCRIPTION
**Pull request by drmateo**

### CODE REVIEW

**Comment by qgallouedec** [comment]
Hi, can you elaborate about the need behind that PR?

**Comment by drmateo** [comment]
Sure!  This patch is more a style problem that must be resolved as I purpose. The two main reasons are: 
- Attribute np_random is a member attribute of the Task class, so for the sake of readability this attribute must be first declared in Task.__init__ method, instead of in another secondary class as it was the case before this patch; it was first declared in reset method of  RobotTaskEnv. That made a little difficult to track the origin of att Task.np_random
- The other problem is, that this patch resolves problems with syntax check tools such as Pylance. Mainly, inability to resolve attributo Task.np_random.

File: panda_gym/envs/core.py [hunk 1]
```diff
@@ -165,10 +165,19 @@ class Task(ABC):
     def __init__(self, sim: PyBullet) -> None:
         self.sim = sim
         self.goal = None
+        self.np_random = Any
```
Comment (qgallouedec) [inline panda_gym/envs/core.py:168]:
```python
self.np_random: np.random.Generator
```

instead

**Comment by qgallouedec** [comment]
Sound fair.
However, I would rather go for a `set_np_random(np_random)` instead of the `set_random_seed(seed)`. By the way, this method shouldn't be protected, as it its intended to be called outside the class.

**Review by qgallouedec — CHANGES_REQUESTED** [review]

**Pull request closed without merging**